File: Make the array beautiful[ 08 April 2023]/Solution.cpp
//  >> C++ CODE <<

vector<int> makeBeautiful(vector<int> arr) {
        
         vector<int> stack;
         for(int i : arr) {
             if(stack.size() != 0 && ((stack[0] >= 0 && i < 0) || (stack[0] < 0 && i >= 0))) {
                 stack.pop_back();
             } else {
                 stack.push_back(i);
             }
         }
         return stack;
        
    }

File: Make the array beautiful[ 08 April 2023]/solution.py
#    >> PYTHON CODE <<

from typing import List

class Solution:
    def makeBeautiful(self, arr: List[int]) -> List[int]:
        stack=[]
        
        for i in arr:
            if len(stack) is not 0 and ((stack[-1]>=0 and i<0) or (stack[-1]<0 and i>=0)):
                stack.pop()
            else:
                stack.append(i)
        
        return stack